quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -237,10 +243,26 @@ private void configureDatabase(ExtensionContext context) {
         WithDatabase database = context.getTestClass().orElse(Object.class).getDeclaredAnnotation(WithDatabase.class);
 
         if (database != null) {
-            configureDevServices();
-            setProperty("kc.db", database.alias());
-            // databases like mssql are very strict about password policy
-            setProperty("kc.db-password", "Password1!");
+            if (dist == null) {
+                configureDevServices();
+                setProperty("kc.db", database.alias());
+                setProperty("kc.db-password", DatabaseContainer.DEFAULT_PASSWORD);
+            } else {
+                databaseContainer = new DatabaseContainer(database.alias());
+
+                databaseContainer.start();
+
+                dist.setProperty("db", database.alias());
+                dist.setProperty("db-username", databaseContainer.getUsername());
+                dist.setProperty("db-password", databaseContainer.getPassword());
+                dist.setProperty("db-url", databaseContainer.getJdbcUrl());
+
+                dist.start(List.of("build"));
+
+                if (database.removeBuildOptionsAfterBuild()) {
```

Comment:
Reflecting the encoding `removeBuildOptionsAfterBuild` -> `removeDBOptionAfterBuild`, wdyt?

Reply:
> it should be generic enough to remove any build option set when configuring the database.

fine for me if that's the behavior, in this case, we are only removing the `db` property, we can use the `PropertyMapper` and iterate all the build time properties.

Reply:
You make me wonder if this option makes more sense at the `DistributionTest` annotation instead because it is not specific to database tests.

I've updated the PR and moved this option to reflect that. Looks better to me now.